clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp

```diff
@@ -735,11 +736,15 @@ wrapDeviceImages(ArrayRef<std::unique_ptr<MemoryBuffer>> Buffers,
 }
 
 Expected<SmallVector<std::unique_ptr<MemoryBuffer>>>
-bundleOpenMP(ArrayRef<OffloadingImage> Images) {
+bundleOpenMP(SmallVectorImpl<OffloadingImage> &Images) {
```

Comment:
Why does the container need to be mutable now?

Reply:
It's mutable because we modify it inside `bundleOpenMP` if the vendor is `Intel`, where we call `containerizeOpenMPSPIRVImage` to put the device image into an ELF (which we can't do using clang because there's no SPIR-V linker). I agree it's not ideal to have to modify the image here, if you have a better spot to do that ELF packaging, let me know. Thanks

Reply:
But this function's job is to return a new list of binary files after serializing the input, can't it just be put in the output?

Reply:
Somehow I convinced myself it might be more performant to do it this way, but I'm not so sure that's true, let me try to just modified the serialized input.